tests/suit/tests_suite_test.go

```diff
@@ -0,0 +1,101 @@
+	goCLIPath         = flag.String("gocli-path", "cli.sh", "The path to cli script")
+	dockerPrefix      = flag.String("docker-prefix", "", "The docker host:port")
+	dockerTag         = flag.String("docker-tag", "", "The docker tag")
+)
+
+// forkliftFailHandler call ginkgo.Fail with printing the additional information
+func forkliftFailHandler(message string, callerSkip ...int) {
+	if len(callerSkip) > 0 {
+		callerSkip[0]++
+	}
+	ginkgo.Fail(message, callerSkip...)
+}
+
+func TestTests(t *testing.T) {
+	defer GinkgoRecover()
+	RegisterFailHandler(forkliftFailHandler)
+	BuildTestSuite()
+	RunSpecsWithDefaultAndCustomReporters(t, "Tests Suite", reporters.NewReporters())
+}
+
+// To understand the order in which things are run, read http://onsi.github.io/ginkgo/#understanding-ginkgos-lifecycle
+// flag parsing happens AFTER ginkgo has constructed the entire testing tree. So anything that uses information from flags
+// cannot work when called during test tree construction.
+func BuildTestSuite() {
+	BeforeSuite(func() {
+		fmt.Fprintf(ginkgo.GinkgoWriter, "Reading parameters\n")
+		// Read flags, and configure client instances
+		framework.ClientsInstance.KubectlPath = *kubectlPath
+		framework.ClientsInstance.OcPath = *ocPath
+		framework.ClientsInstance.ForkliftInstallNs = *forkliftInstallNs
+		framework.ClientsInstance.KubeConfig = *kubeConfig
+		framework.ClientsInstance.KubeURL = *kubeURL
+		framework.ClientsInstance.GoCLIPath = *goCLIPath
+		framework.ClientsInstance.DockerPrefix = *dockerPrefix
+		framework.ClientsInstance.DockerTag = *dockerTag
+
+		fmt.Fprintf(ginkgo.GinkgoWriter, "Kubectl path: %s\n", framework.ClientsInstance.KubectlPath)
+		fmt.Fprintf(ginkgo.GinkgoWriter, "OC path: %s\n", framework.ClientsInstance.OcPath)
+		fmt.Fprintf(ginkgo.GinkgoWriter, "Forklift install NS: %s\n", framework.ClientsInstance.ForkliftInstallNs)
+		fmt.Fprintf(ginkgo.GinkgoWriter, "Kubeconfig: %s\n", framework.ClientsInstance.KubeConfig)
+		fmt.Fprintf(ginkgo.GinkgoWriter, "KubeURL: %s\n", framework.ClientsInstance.KubeURL)
+		fmt.Fprintf(ginkgo.GinkgoWriter, "GO CLI path: %s\n", framework.ClientsInstance.GoCLIPath)
+		fmt.Fprintf(ginkgo.GinkgoWriter, "DockerPrefix: %s\n", framework.ClientsInstance.DockerPrefix)
+		fmt.Fprintf(ginkgo.GinkgoWriter, "DockerTag: %s\n", framework.ClientsInstance.DockerTag)
+
+		restConfig, err := framework.ClientsInstance.LoadConfig()
+		if err != nil {
+			// Can't use Expect here due this being called outside of an It block, and Expect
+			// requires any calls to it to be inside an It block.
+			ginkgo.Fail("ERROR, unable to load RestConfig")
+		}
+		framework.ClientsInstance.RestConfig = restConfig
+		// clients
+		kcs, err := framework.ClientsInstance.GetKubeClient()
+		if err != nil {
+			ginkgo.Fail(fmt.Sprintf("ERROR, unable to create K8SClient: %v", err))
+		}
+		framework.ClientsInstance.K8sClient = kcs
+
+		crClient, err := framework.ClientsInstance.GetCrClient()
```

Comment:
if the crClient is the runtime client, why not calling it runtimeClient? what's `cr` means?